### PR TITLE
suggest `MAX` constant if -1 is assigned to unsigned type

### LINKED ISSUE #76413: Error message on casting negative numbers to usize is quite unhelpful
Code:
```rust
fn syscall(nr: usize) -> usize {
    match nr {
        1 => (-1) as usize,
        _ => -1 as usize,
    }
}

fn syscall_i(nr: usize) -> isize {
    match nr {
        1 => (-1) as isize,
        _ => -1 as isize,
    }
}

fn main() {
    let rc = syscall(42);
    println!("result: {}", rc);
    
    let rc = syscall_i(42);
    println!("result: {}", rc);
}
```
Playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=04afbf8753fe268ea47b7b3fae6f897c

Error message:
```
error[E0600]: cannot apply unary operator `-` to type `usize`
 --> src/main.rs:3:14
  |
3 |         1 => (-1) as usize,
  |              ^^^^ cannot apply unary operator `-`
  |
  = note: unsigned values cannot be negated

error[E0600]: cannot apply unary operator `-` to type `usize`
 --> src/main.rs:4:14
  |
4 |         _ => -1 as usize,
  |              ^^ cannot apply unary operator `-`
  |
  = note: unsigned values cannot be negated
```

Somehow, in one case this `1` in the `-1` expression is unsigned, but in the other case it is signed and everything is fine.

### LINKED ISSUE #77416: [ER] Propose the use of XX::MAX instead of -1 for unsigned integers
Another of those little enhancement suggestions. This code:

```rust
fn main() {
    let x: u32 = -1;
}
```

With rustc 1.48.0-nightly gives this error message:

```rust
error[E0600]: cannot apply unary operator `-` to type `u32`
 --> ...\temp.rs:2:18
  |
2 |     let x: u32 = -1;
  |                  ^^ cannot apply unary operator `-`
  |
  = note: unsigned values cannot be negated
```

I suggest to modify the note, to suggest that u32::MAX may be desired instead, because in C code often you see people use -1 to get the max unsigned value.

### PR DESCRIPTION
**Pull request by euclio**
Fixes #76413.
Fixes #77416.

### CODE REVIEW

File: compiler/rustc_typeck/src/check/op.rs [hunk 1]
```diff
@@ -673,6 +675,26 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
...
+                                            node: ast::LitKind::Int(1, _),
+                                            ..
+                                        }),
+                                    ..
+                                },
+                            ) = ex.kind
+                            {
+                                err.span_suggestion(
+                                    ex.span,
+                                    &format!("obtain the maximum `{}` value with its `MAX` associated constant", actual),
```
Comment (euclio) [inline compiler/rustc_typeck/src/check/op.rs:693]: Not sure if "its" is the right word here. "the" might work better.

Comment (Lonami) [reply compiler/rustc_typeck/src/check/op.rs:693]: I think "its" is the right word here, since you want to talk about the possession of a "it".

Comment (davidtwco) [inline compiler/rustc_typeck/src/check/op.rs:693]:
```suggestion
                                    &format!("you might have meant to use `{}`'s maximum value", actual),
```

**Review by davidtwco — CHANGES_REQUESTED** [review]
Implementation wise, this looks great. I'd like to see the message changed slightly, I don't think we've got a definitive style guide here but it currently reads slightly awkwardly to me.

File: compiler/rustc_typeck/src/check/op.rs (hunk 1 above)
Comment (euclio) [reply compiler/rustc_typeck/src/check/op.rs]: Fixed, but tweaked slightly.

**Review by davidtwco — APPROVED** [review]

**Pull request merged**